src/core/math_func.hpp

```diff
@@ -159,6 +159,18 @@ static inline uint16 ClampToU16(const uint64 a)
 	return static_cast<uint16>(std::min(a, static_cast<uint64>(UINT16_MAX)));
 }
 
+/**
+ * Underflow safe unsigned subtraction
+ * @param a First number
+ * @param b Second number
+ * @return a - b, or 0 if it would underflow
+ */
+static inline uint UnderflowSafeSub(uint a, uint b)
+{
+	if (b < a) return 0;
+	return a - b;
```

Comment:
Shouldn't it be `if (b > a)`?

Reply:
He is testing us, I am telling you :) Just to see if the reviewers are awake .. I clearly wasn't :P